BlockBlock/Assets/Scripts/GameManager.cs

```diff
@@ -1,359 +1,374 @@
+        }
+
+        // Clear cols
+        foreach(int col in colsToClear)
+        {
+            ClearLine(col, false);
+        }
+
+        // Clear hashsets used to indicate which rows/cols to check
+        rowsToCheck.Clear();
+        colsToCheck.Clear();
+    }
+
+    // Clears row/col
+    private void ClearLine(int index, bool rowFlag)
+    {
+        // Clear row if flag is set
+        if (rowFlag)
+        {
+            for (int x = 0; x < GRID_WIDTH; x++)
+            {
+                if (grid[x, index] != null)
+                {
+                    // set trigger to transition to the destroy animation
+                    grid[x, index].gameObject.GetComponent<Animator>().SetTrigger("DestroyAnimation");
+
+                    // wait 1 second to allow the animation to finish before destroying the block piece
+                    Destroy(grid[x, index].gameObject, 1f);
+                    grid[x, index] = null;
+                }
+            }
+        }
+        // Otherwise clear col
+        else
+        {
+            for (int y = 0; y < GRID_HEIGHT; y++)
+            {
+                if (grid[index, y] != null)
+                {
+                    // set trigger to transition to the destroy animation
+                    grid[index, y].gameObject.GetComponent<Animator>().SetTrigger("DestroyAnimation");
+
+                    // wait 1 second to allow the animation to finish before destroying the block piece
+                    Destroy(grid[index, y].gameObject, 1f);
+                    grid[index, y] = null;
+                }
+            }
+        }
+    }
+
+    // Checks if the row/col is full
+    private bool IsLineFull(int index, bool rowFlag)
+    {
+        // Check row if flag is set
+        if (rowFlag)
+        {
+            for (int x = 0; x < GRID_WIDTH; x++)
+            {
+                if (grid[x, index] == null)
+                {
+                    return false;
+                }
+            }
+        }
+        // Otherwise check col
+        else
+        {
+            for (int y = 0; y < GRID_HEIGHT; y++)
+            {
+                if (grid[index, y] == null)
+                {
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    }
+
+    private void SpawnBlocks()
+    {
+        for (int i = 0; i < NUM_SPAWN; i++)
+        {
+            // Get a random block
+            Block block = blocks[Random.Range(0, blocks.Length)];
+
+            // Spawn the block at the respective spawn position
+            Block toSpawn = Instantiate(block, spawnPositions[i], Quaternion.identity);
+            spawnedBlocks[i] = toSpawn;
+            toSpawn.transform.SetParent(pieces);
+
+            // Apply random rotation to block
+            RotateBlock(toSpawn);
+
+            // iterate through each of the block's child pieces to trigger their spawn animations
+            foreach(Transform blockPiece in toSpawn.transform)
```

Comment:
this is where we iterate through each of the block's child pieces to trigger their spawn animations